Automattic/wp-calypso
# Reader Onboarding: Reload page Stream when SubscribeModal closes

Related to p1730319313841119-slack-C03NLNTPZ2T

## Proposed Changes

* This PR introduces `READER_STREAMS_CLEAR`
* It uses `READER_STREAMS_CLEAR` to clear and reload the Reader stream anytime we close the SubscribeModal.



Before | After
--|--
<video src="https://github.com/user-attachments/assets/bd9f08f0-5686-4e75-8167-f3c8dceefd28" /> | <video src="https://github.com/user-attachments/assets/d4e4b1e5-179d-498c-83ea-fb8081596a44" /> 

## Why are these changes being made?


* When users follow and unfollow blogs during Reader Onboarding, they should see the results of those actions on their stream right away so they're not left wondering if anything actually happened.

## Testing Instructions



* Go to /read?flags=reader/onboarding on a new account with a verified email
* Follow some blogs using Reader Onboarding `SubscribeModal` and close the modal by any means to see if the Stream gets updated.
* Make sure clicking "Continue" on the SubscribeModal also updates the Stream.
* Consider if there could be regressions by introducing the `READER_STREAMS_CLEAR` action.

## Pre-merge Checklist
